Framework/Algorithms/src/ReflectometryReductionOne2.cpp

```diff
@@ -362,5 +750,396 @@ ReflectometryReductionOne2::convertToQ(MatrixWorkspace_sptr inputWS) {
+* values will all be zero.
+*
+* @return : a 1D workspace where y values are all zero
+*/
+MatrixWorkspace_sptr ReflectometryReductionOne2::constructIvsLamWS(
+    MatrixWorkspace_sptr detectorWS, const std::vector<size_t> &detectors) {
+  // Construct the new workspace from the centre pixel (where twoThetaR is
+  // defined)
+  auto cropWorkspaceAlg = this->createChildAlgorithm("CropWorkspace");
+  cropWorkspaceAlg->initialize();
+  cropWorkspaceAlg->setProperty("InputWorkspace", detectorWS);
+  cropWorkspaceAlg->setProperty(
+      "StartWorkspaceIndex", static_cast<int>(twoThetaRDetectorIdx(detectors)));
+  cropWorkspaceAlg->setProperty(
+      "EndWorkspaceIndex", static_cast<int>(twoThetaRDetectorIdx(detectors)));
+  cropWorkspaceAlg->execute();
+  MatrixWorkspace_sptr ws = cropWorkspaceAlg->getProperty("OutputWorkspace");
+
+  // Get the max/min wavelength of region of interest
+  const double lambdaMin = getProperty("WavelengthMin");
+  const double lambdaMax = getProperty("WavelengthMax");
+
+  // Get the max and min values of the projected (virtual) lambda range
+  double lambdaVMin = 0.0;
+  double lambdaVMax = 0.0;
+  double dummy = 0.0;
+
+  const size_t spIdxMin = detectors.front();
+  const double twoThetaMin = getDetectorTwoTheta(m_spectrumInfo, spIdxMin);
+  const double bTwoThetaMin =
+      getDetectorTwoThetaRange(m_spectrumInfo, spIdxMin);
+  // For bLambda, use the average bin size for this spectrum
+  auto xValues = detectorWS->x(spIdxMin);
+  double bLambda = (xValues[xValues.size() - 1] - xValues[0]) /
+                   static_cast<int>(xValues.size());
+  getProjectedLambdaRange(lambdaMax, twoThetaMin, bLambda, bTwoThetaMin,
+                          detectors, lambdaVMin, dummy);
+
+  const size_t spIdxMax = detectors.back();
+  const double twoThetaMax = getDetectorTwoTheta(m_spectrumInfo, spIdxMax);
+  const double bTwoThetaMax =
+      getDetectorTwoThetaRange(m_spectrumInfo, spIdxMax);
+  xValues = detectorWS->x(spIdxMax);
+  bLambda = (xValues[xValues.size() - 1] - xValues[0]) /
+            static_cast<int>(xValues.size());
+  getProjectedLambdaRange(lambdaMin, twoThetaMax, bLambda, bTwoThetaMax,
+                          detectors, dummy, lambdaVMax);
+
+  if (lambdaVMin > lambdaVMax) {
+    std::swap(lambdaVMin, lambdaVMax);
+  }
+
+  // Use the same number of bins as the input
+  const int origNumBins = static_cast<int>(detectorWS->blocksize());
+  const double binWidth = (lambdaMax - lambdaMin) / origNumBins;
+  std::stringstream params;
+  params << lambdaVMin << "," << binWidth << "," << lambdaVMax;
+
+  auto rebinAlg = this->createChildAlgorithm("Rebin");
+  rebinAlg->initialize();
+  rebinAlg->setProperty("InputWorkspace", ws);
+  rebinAlg->setProperty("PreserveEvents", false);
+  rebinAlg->setProperty("Params", params.str());
+  rebinAlg->execute();
+  ws = rebinAlg->getProperty("OutputWorkspace");
+
+  for (int i = 0; i < static_cast<int>(ws->blocksize()); ++i) {
```

Comment:
This was a roundabout way of creating an empty workspace for the output. I've changed the way it gets created.